plugins/chat/assets/javascripts/discourse/models/chat-channel.js

```diff
@@ -105,6 +115,53 @@ export default class ChatChannel extends RestModel {
     return this.currentUserMembership.following;
   }
 
+  set details(details) {
+    this.canDeleteOthers = details.can_delete_others ?? false;
+    this.canDeleteSelf = details.can_delete_self ?? false;
+    this.canFlag = details.can_flag ?? false;
+    this.canModerate = details.can_moderate ?? false;
+    if (details.can_load_more_future !== undefined) {
+      this.canLoadMoreFuture = details.can_load_more_future;
+    }
+    if (details.can_load_more_past !== undefined) {
+      this.canLoadMorePast = details.can_load_more_past;
+    }
+    this.userSilenced = details.user_silenced ?? false;
+    this.status = details.channel_status ?? "open";
+    this.channelMessageBusLastId = details.channel_message_bus_last_id;
+  }
+
+  clearMessages() {
```

Comment:
As I said earlier, we will likely also need all this kind of stuff on the Thread JS model, its good we have a framework for it now. Maybe this could be some mixin class we can extend so both can use it, since these 6 functions are pretty generic? Something like `HasChatMessages`